integration-tests/sql/src/test/java/org/apache/camel/quarkus/component/sql/it/SqlTest.java

```diff
@@ -66,4 +78,169 @@ public void testSqlStoredComponent() {
+
+        postMap("/sql/toDirect/transacted", CollectionHelper.mapOf(SqlConstants.SQL_QUERY,
+                "insert into projects values (6, 'Transacted', 'ASF', false)",
+                "rollback", true))
+                        .statusCode(200)
+                        .body(is("java.lang.Exception:forced Exception"));
+
+        postMap("/sql/toDirect/transacted",
+                CollectionHelper.mapOf(SqlConstants.SQL_QUERY, "select * from projects where project = 'Transacted'"))
+                        .statusCode(200)
+                        .body("size()", is(1));
+    }
+
+    @Test
+    public void testDefaultErrorCode() throws InterruptedException {
+        postMap("/sql/toDirect/transacted", CollectionHelper.mapOf(SqlConstants.SQL_QUERY, "select * from NOT_EXIST order id"))
+                .statusCode(200)
+                .body(startsWith("org.springframework.jdbc.BadSqlGrammarException"));
+    }
+
+    @Test
+    public void testIdempotentRepository() {
+        // add value with key 1
+        postMapWithParam("/sql/toDirect/idempotent",
+                "body", "one",
+                CollectionHelper.mapOf("messageId", "1"))
+                        .statusCode(200);
+
+        // add value with key 2
+        postMapWithParam("/sql/toDirect/idempotent",
+                "body", "two",
+                CollectionHelper.mapOf("messageId", "2"))
+                        .statusCode(200);
+
+        // add same value with key 3
+        postMapWithParam("/sql/toDirect/idempotent",
+                "body", "three",
+                CollectionHelper.mapOf("messageId", "3"))
+                        .statusCode(200);
+
+        // add another value with key 1 -- this one is supposed to be skipped
+        postMapWithParam("/sql/toDirect/idempotent",
+                "body", "four",
+                CollectionHelper.mapOf("messageId", "1"))
+                        .statusCode(200);
+
+        // get all values from the result map
+        await().atMost(5, TimeUnit.SECONDS).until(() -> (Iterable<? extends String>) RestAssured
+                .get("/sql/get/results/idempotentRoute").then().extract().as(List.class),
+                containsInAnyOrder("one", "two", "three"));
+    }
+
+    @Test
+    @Disabled //see https://github.com/apache/camel-quarkus/issues/2693
+    public void testAggregationRepository() {
+        postMapWithParam("/sql/toDirect/aggregation", "body", "A", CollectionHelper.mapOf("messageId", "123"))
+                .statusCode(200);
+
+        postMapWithParam("/sql/toDirect/aggregation", "body", "B", CollectionHelper.mapOf("messageId", "123"))
+                .statusCode(200);
+
+        postMapWithParam("/sql/toDirect/aggregation", "body", "C", CollectionHelper.mapOf("messageId", "123"))
+                .statusCode(200);
+
+        postMapWithParam("/sql/toDirect/aggregation", "body", "D", CollectionHelper.mapOf("messageId", "123"))
+                .statusCode(200);
```

Comment:
Not packing the RestAssured code into a method would make it easier to understand what this actually does. Maybe it is just me? What do others think?

Reply:
I tried to keep all parametetrs (like url, body, params) in the main call (not helper method) - from my POV it seems more understandable and shows that all the calls are the same.
But I don't have any problem with keeping restassured  code intact...

Reply:
Plz. do not change anything, maybe it is just me :)